### PR TITLE
NOISSUE - Remove domain from token

### PR DESCRIPTION
**Pull request by nyagamunene**
# What type of PR is this?
This is a refactor because it changes the following functionality: it removes domain from token since it is passed with the URL.

## What does this do?
It removes domain from token.

## Which issue(s) does this PR fix/relate to?

- Related Issue: N/A
- Resolves: N/A

## Have you included tests for your changes?
Yes.

## Did you document any new/modified feature?
No

### CODE REVIEW

**Review by arvindh123 — DISMISSED** [review]

File: api/openapi/invitations.yml [hunk 1]
```diff
@@ -48,6 +48,8 @@ paths:
           description: Failed due to malformed JSON.
         "401":
           description: Missing or invalid access token provided.
+        "403":
+          description: Unauthorized access the domain ID.
```
Comment (felixgateru) [inline api/openapi/invitations.yml]: Could you improve the grammar on this

**Review by felixgateru — CHANGES_REQUESTED** [review]

File: invitations/api/transport.go [hunk 2]
```diff
@@ -155,16 +153,14 @@ func decodeAcceptInvitationReq(_ context.Context, r *http.Request) (interface{},
...
+		token: apiutil.ExtractBearerToken(r),
 	}, nil
 }
 
 func decodeInvitationReq(_ context.Context, r *http.Request) (interface{}, error) {
 	req := invitationReq{
-		token:    apiutil.ExtractBearerToken(r),
-		userID:   chi.URLParam(r, "user_id"),
-		domainID: chi.URLParam(r, "domainID"),
+		token:  apiutil.ExtractBearerToken(r),
```
Comment (felixgateru) [inline invitations/api/transport.go]: Please remove the token from this section, I dont think it is used anymore

**Review by felixgateru — CHANGES_REQUESTED** [review]

**Review by felixgateru — DISMISSED** [review]
LGTM

**Review by arvindh123 — DISMISSED** [review]
CLI suggestion: (This can be done in separate PR)

We can add context management to the CLI, allowing users to manage multiple sessions with `domain_id` and `token`.

 Users can:
Add a context with `mg-cli add-context <DomainID> <token> <contextName>`
List all contexts using `mg-cli list-context`
Switch contexts with `mg-cli use-context <contextName>`

This setup helps manage multiple sessions seamlessly.


LGTM

**Comment by dborovcanin** [comment]
> CLI suggestion: (This can be done in separate PR)
> 
> We can add context management to the CLI, allowing users to manage multiple sessions with `domain_id` and `token`.
> 
> Users can: Add a context with `mg-cli add-context <DomainID> <token> <contextName>` List all contexts using `mg-cli list-context` Switch contexts with `mg-cli use-context <contextName>`
> 
> This setup helps manage multiple sessions seamlessly.
> 
> LGTM

I agree. I have created a reference issue. I was thinking of something similar recently, like `nats` CLI command, or even `kubectl`.

File: cli/provision.go [hunk 3]
```diff
@@ -166,8 +166,8 @@ var cmdProvision = []cobra.Command{
 				return
 			}
 
-			// domain login
-			ut, err = sdk.CreateToken(mgxsdk.Login{Identity: user.Credentials.Identity, Secret: user.Credentials.Secret, DomainID: domain.ID})
+			// Create token
```
Comment (dborovcanin) [inline cli/provision.go]: Remove this comment, it's obvious.

File: cli/users.go [hunk 4]
```diff
@@ -91,13 +91,13 @@ var cmdUsers = []cobra.Command{
 		},
 	},
 	{
-		Use:   "token <username> <password> [<domainID>]",
+		Use:   "token <username> <password>]",
```
Comment (dborovcanin) [inline cli/users.go]: Fix format (an extra `]`).

File: cli/users.go [hunk 5]
```diff
@@ -120,22 +117,18 @@ var cmdUsers = []cobra.Command{
 		},
 	},
 	{
-		Use:   "refreshtoken <token> [<domainID>]",
+		Use:   "refreshtoken <token>]",
```
Comment (dborovcanin) [inline cli/users.go]: Same.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**